app/src/main/java/com/monkeyteam/chimpagne/MainActivity.kt

```diff
@@ -132,13 +133,38 @@ class MainActivity : ComponentActivity() {
               MyEventsScreen(navObject = navActions, myEventsViewModel = myEventsViewModel)
             }
             composable(Route.VIEW_DETAIL_EVENT_SCREEN + "/{EventID}") { backStackEntry ->
+              ViewDetailEventScreen(
+                  navObject = navActions,
+                  eventViewModel =
+                      viewModel(
+                          factory =
+                              EventViewModelFactory(
+                                  backStackEntry.arguments?.getString("EventID"), database)))
+            }
+            composable(Route.JOIN_EVENT_SCREEN) {
               val eventViewModel: EventViewModel =
-                  viewModel(
-                      factory =
-                          EventViewModelFactory(
-                              backStackEntry.arguments?.getString("EventID"), database))
-              eventViewModel.fetchEvent()
-              ViewDetailEventScreen(navObject = navActions, eventViewModel = eventViewModel)
+                  viewModel(factory = EventViewModelFactory(null, database))
+              /*
```

Comment:
As discussed in the previous SCRUM meeting, this kind of bloc of commented code should not be merged in main.